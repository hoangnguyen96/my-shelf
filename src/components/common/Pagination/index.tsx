"use client";

import { Dispatch, memo, SetStateAction } from "react";
import { ChevronLeftIcon, ChevronRightIcon } from "@chakra-ui/icons";
import { Flex, IconButton } from "@chakra-ui/react";
import { Button } from "..";

interface PaginationProps {
  pagination: number;
  totalPages: number;
  setPagination: Dispatch<SetStateAction<number>>;
}

const Pagination = ({
  pagination,
  totalPages,
  setPagination,
}: PaginationProps) => (
  <Flex
    justifyContent="center"
    alignItems="center"
    gap="8px"
    position="absolute"
    left="0"
    right="0"
    bottom="5%"
  >
    <IconButton
      data-testid="prev-pagination"
      width="30px"
      height="30px"
      size="sm"
      variant="outline"
      color="brand.90"
      bgColor="transparent"
      onClick={() => setPagination(pagination - 1)}
      isDisabled={pagination === 0}
      icon={<ChevronLeftIcon />}
      minW="auto"
      aria-label="prev-pagination"
      _focusVisible="none"
      _hover={{ bgColor: "transparent" }}
    />

    {Array.from({ length: totalPages }, (_, index) => (
      <Button
        key={index}
        data-testid="click-pagination"
        width="30px"
        height="30px"
        text={`${index + 1}`}
        variant="outline"
        size="sm"
        minW="auto"
        isActive={pagination === index}
        onClick={() => setPagination(index)}
      />
    ))}

    <IconButton
      data-testid="next-pagination"
      size="sm"
      width="30px"
      height="30px"
      variant="outline"
      color="brand.90"
      bgColor="transparent"
      minW="auto"
      onClick={() => setPagination(pagination + 1)}
      isDisabled={pagination === totalPages - 1}
      icon={<ChevronRightIcon />}
      aria-label="next-pagination"
      _focusVisible="none"
      _hover={{ bgColor: "transparent" }}
    />
  </Flex>
);

export default memo(Pagination);
