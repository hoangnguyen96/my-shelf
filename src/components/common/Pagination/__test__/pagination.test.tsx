import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import Pagination from "..";

describe("Pagination", () => {
  const props = {
    pagination: 1,
    totalPages: 2,
    setPagination: jest.fn(),
  };

  it("Should render correctly snapshot", () => {
    expect(render(<Pagination {...props} />)).toMatchSnapshot();
  });

  it("Should handle previous pagination", () => {
    const { getByTestId } = render(<Pagination {...props} />);

    const button = getByTestId("prev-pagination");
    fireEvent.click(button);
  });

  it("Should handle click pagination", () => {
    const { getAllByTestId } = render(<Pagination {...props} />);

    const button = getAllByTestId("click-pagination");
    fireEvent.click(button[1]);
  });

  it("Should handle next pagination", () => {
    const { getByTestId } = render(<Pagination {...props} totalPages={4} />);

    const button = getByTestId("next-pagination");
    fireEvent.click(button);
  });
});
