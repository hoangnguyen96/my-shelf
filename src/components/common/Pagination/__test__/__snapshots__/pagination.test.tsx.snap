// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-9d3smw"
      >
        <button
          aria-label=""
          class="chakra-button css-7baeyj"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="chakra-button css-klz9de"
          type="button"
        >
          1
        </button>
        <button
          aria-label=""
          class="chakra-button css-7baeyj"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-9d3smw"
    >
      <button
        aria-label=""
        class="chakra-button css-7baeyj"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="chakra-button css-klz9de"
        type="button"
      >
        1
      </button>
      <button
        aria-label=""
        class="chakra-button css-7baeyj"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
